python/tchannel/tornado/peer.py

```diff
@@ -188,18 +205,23 @@ def choose(self, hostport=None, score_threshold=None):
             If specified, Peers with a score equal to or below this will be
             ignored. Defaults to the value specified when the PeerGroup was
             initialized.
+        :param blacklist:
+            Peers on the blacklist won't be chosen.
         :returns:
             A Peer that matches all the requested criteria or None if no such
             Peer was found.
         """
-        if hostport:
+
+        blacklist = blacklist or set()
+        if hostport and hostport not in blacklist:
             return self.get(hostport)
```

Comment:
This is completely breaking the semantics of how the method works. If `hostport` is given, it should *always* returns that peer. Blacklist can't apply.

On the Node side, if a hostport is specified, they just don't do retries for that request. We should do the same.